docs/source/examples/tips_and_tricks.rst

```diff
@@ -0,0 +1,492 @@
+
+        def __call__(self, address):
+            """
+            Convert address to embedding vectors
+            :param address: The address to convert
+            :return: The embeddings vectors
+            """
+            embeddings = []
+            for word in address.split():
+                embeddings.append(self.embedding_model[word])
+            return embeddings
+
+    embedding_model = EmbeddingVectorizer()
+
+We also need a vectorizer to convert the address tag (e.g. StreeNumber, StreetName) into categorical values. So we will use a Vectorizer class that can use the embedding vectorizer and convert the address tag.
+
+.. code-block:: python
+
+    class Vectorizer:
+        def __init__(self, dataset, embedding_model):
+            self.data = dataset
+            self.embedding_model = embedding_model
+            self.tags_set = {
+                "StreetNumber": 0,
+                "StreetName": 1,
+                "Unit": 2,
+                "Municipality": 3,
+                "Province": 4,
+                "PostalCode": 5,
+                "Orientation": 6,
+                "GeneralDelivery": 7
+            }
+
+        def __len__(self):
+            # for the dataloader
+            return len(self.data)
+
+        def __getitem__(self, item):
+            data = self.data[item]
+            address = data[0]
+            address_vector = self.embedding_model(address)
+
+            tags = data[1]
+            idx_tags = self._convert_tags_to_idx(tags)
+
+            return address_vector, idx_tags
+
+        def _convert_tags_to_idx(self, tags):
+            idx_tags = []
+            for tag in tags:
+                idx_tags.append(self.tags_set[tag])
+            return idx_tags
+
+
+.. code-block:: python
+
+    train_data_vectorize = Vectorizer(train_data, embedding_model)
+    valid_data_vectorize = Vectorizer(valid_data, embedding_model)
+    test_data_vectorize = Vectorizer(test_data, embedding_model)
+
+DataLoader
+^^^^^^^^^^
+
+Now, since all the addresses are not of the same size, it is impossible to batch size them since all elements of a tensor must have the same lengths. But there a trick, padding!
+
+The idea is simple. We will add *empty* tokens at the ends of a sequence up to the longest one in a batch. At the moment of evaluating the loss, that tokens will be skip using a ignore value (`0`). That way, we can pad and pack the sequence to minimize the training time (read `this <https://stackoverflow.com/questions/51030782/why-do-we-pack-the-sequences-in-pytorch>`_ good explanation of why we pad and pack sequence).
+
+Also, due to how the F1 Score to compute is done, we will need a mask to ignore the paddings elements when calculating the metric. The mask value will be set to (`-100`) and will be used only at running time (see the `documentation <https://poutyne.org/metrics.html#poutyne.framework.metrics.FBeta.forward>`_ for more details).
```

Comment:
The values in the mask are supposed to be zeros and ones, setting them to -100 will result in some kind of bug... Code to modify below too.

Reply:
Yeah, I've misexplained. I will rewrite.

Reply:
4b2c42d